lib/identity/hostdata/config_builder.rb

```diff
@@ -68,13 +69,48 @@ def add(key, type: :string, allow_nil: false, enum: nil, options: {})
 
         key_types[key] = type
 
+        converted_value = convert!(
+          key: key,
+          value: value,
+          type: type,
+          allow_nil: allow_nil,
+          enum: enum,
+          options: options,
+        )
+
+        @written_env[key] = converted_value.freeze
+      end
+
+      def add_ssm(prop_name, ssm_name, type: :string, allow_nil: false, enum: nil, options: {})
+        raw_value = load_ssm_value(ssm_name)
+
+        key_types[prop_name] = type
+
+        converted_value = if block_given?
```

Comment:
What's the block usage? Should we document it?

Reply:
see spec, it allows custom post-processing on the secrets. I'll update documentation

Reply:
documented in 92c6221c6078e2c0c24ca988e53657aaa4ec383b